File: 551.py
class Solution:
    def checkRecord(self, s: str) -> bool:
        consecLCount = 0
        prevIsL = False
        aCount = 0
        for c in s:
            if c == 'A':
                aCount += 1
                if aCount == 2:
                    return False
                prevIsL = False
            elif c == 'L':
                if prevIsL:
                    consecLCount += 1
                    if consecLCount == 3:
                        return False
                else:
                    prevIsL = True
                    consecLCount = 1
            else:
                prevIsL = False
        return True

File: sol/551.cpp
struct Solution {
    bool checkRecord(const string& s) {
        bool prevIsL = false;
        short consecLCount = 0, aCount = 0;
        for (auto it = s.cbegin(); it != s.cend(); ++it){
            if (*it == 'L'){
                if (prevIsL){
                    if (consecLCount == 2){
                        return false;
                    }
                    consecLCount += 1;
                } else {
                    consecLCount = 1;
                    prevIsL = true;
                }
            } else {
                if (*it == 'A'){
                    if (aCount == 1){
                        return false;
                    }
                }
                prevIsL = false;
            }
        }
        return true;
    }
};
